src/robot/running/namespace.py

```diff
@@ -287,28 +287,26 @@ def _raise_no_keyword_found(self, name, recommend=True):
         else:
             raise KeywordError(message)
 
-    def _get_runner(self, name):
+    def _get_runner(self, name, strip_bdd_prefix=True):
         if not name:
             raise DataError('Keyword name cannot be empty.')
         if not is_string(name):
             raise DataError('Keyword name must be a string.')
-        runner = self._get_bdd_style_runner(name, self.languages.bdd_prefixes)
+        runner = self._get_bdd_style_runner(name) if strip_bdd_prefix else None
         if not runner:
             runner = self._get_runner_from_suite_file(name)
         if not runner and '.' in name:
             runner = self._get_explicit_runner(name)
         if not runner:
             runner = self._get_implicit_runner(name)
-        if not runner:
-            runner = self._get_bdd_style_runner(name, self.languages.bdd_prefixes)
         return runner
 
-    def _get_bdd_style_runner(self, name, prefixes):
+    def _get_bdd_style_runner(self, name):
         parts = name.split()
         for index in range(1, len(parts)):
             prefix = ' '.join(parts[:index]).title()
-            if prefix in prefixes:
-                runner = self._get_runner(' '.join(parts[index:]))
+            if prefix in self.languages.bdd_prefixes:
```

Comment:
Accessing `self.languages.bdd_prefixes` is a little slower than just accessing `prefixes`, but it's impossible to know without profiling does that matter in any real usage. This code is now used with all keyword calls, not only as the last resort as earlier, so performance is important. I have some ideas for performance improvements that I'll test later. Functionality-wise this ought to be ok so I'll merge this PR first.

Reply:
Yes, and there was another thing still on my mind after changing the code after the first review, although I didn't realise this until writing the user documentation. It now states that a keyword that does not include a bdd prefix gets precedence over one that is written in full, including the prefix in the name. I do not expect this to be a practical issue in any way. If you start mixing styles you will be in trouble at one point anyway. For instance, when using the _and_ prefix. But, following the spirit of how we typically treat precedence in Robot, I would expect that the more specific keyword would get precedence over the less specific one.
However, the exact reason for this issue, considering variables as a first word, makes it impossible to find a solution without detecting the variable and/or the prefixes, which was discussed during the first review. So, without seeing any practical benefit, I dismissed the idea. Now, with your latest comment, it seems worth while to reconsider creating a prefix detection function, that can then also be optimised.